### ISSUE TITLE
Simplify GraphQL API: Remove reliance on backend for form generation

### ISSUE THREAD

**Issue by jeffsee55** [tinacms/tinacms#2781]
**Is a Blocker For:** 
- [ ] https://github.com/tinacms/product-roadmap/issues/91

**Summary:**
Now that we have access to the Tina schema on the frontend we can start to remove the logic for generating forms on the backend. There are a few things we do today that rely on this, and we should move the responsibility to the `TinaAdmin` or `TinaSchema` modules:

- [x] Listing collections in the CMS
- [x] Getting fields for a given collection form
- [x] `mutationInfo` provides the mutation string, as well as some info on how to transform the form payload before sending it to the server.
  - `transformDocumentIntoMutationRequestPayload` has some shaky logic in it because it has no knowledge of the schema. Now that we have it client-side, we should be able to remove it's reliance on `mutationInfo`, and instead just look at the schema
  - This has been spun into a separate card: https://github.com/tinacms/tinacms/issues/2822 to address technical debt
- [x] ~~Another natural improvement to make here is to ensure `formifyCallback` is called for admin form generation too.~~
- [ ] Ensure when schema is not provided we throw an error

**Notes:**
This will be a breaking change as users will now have to provide the `schema` as a `TinaCMS` prop

cc @Enigmatical I know we've discussed this, please chime in with any other thoughts.

**Comment by Enigmatical** [comment]
> Another natural improvement to make here is to ensure formifyCallback is called for admin form generation too.

@jeffsee55 and I decided to punt on this criteria as it could potentially have some messy ramifications:
> Right now it just gives you raw access to the form, so a few things like creating global forms or jumping in during the onSubmit, but I think ideally we expose that as functions on the schema and get rid of formifyCallback in the future

> On second thought, I guess that'd be bad if they created a global form on the admin, maybe we can leave that out

**Comment by Enigmatical** [comment]
I had a conversation about this card with @jeffsee55 on Thursday (April 14th) and discussed a way to land it early - mainly to unblock him from finishing the `data` work.

The two remaining goals:
- mutationInfo provides the mutation string, as well as some info on how to transform the form payload before sending it to the server.
  - transformDocumentIntoMutationRequestPayload has some shaky logic in it because it has no knowledge of the schema. Now that we have it client-side, we should be able to remove it's reliance on mutationInfo, and instead just look at the schema
- Ensure when schema is not provided we throw an error

will be spun off into separate cards (separate PRs) with priority on ensuring an error appears when the `schema` is not provided.  The earlier goal is not immediately needed (more of technical debt) and could be addressed at a later point.
